add world schedule and show space events

denisdimitrov opened the pull request:
- Adds world schedule that shows all of the world events
- Renames Timing to Schedule and shows only the space specific events instead of all events

Fixes: https://github.com/sparkletown/internal-sparkle-issues/issues/1624

colinhowe committed:
Merge pull request #2738 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2748 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2749 from sparkletown/staging

Allow bugsnag to override files on upload

denisdimitrov committed:
add world schedule and show space events

denisdimitrov committed:
merge staging into feature/world-schedule

denisdimitrov committed:
replace underscore syntax with camelCase

denisdimitrov committed:
remove duplicated components and adapt new ones

denisdimitrov committed:
fix validation and global time update

denisdimitrov committed:
Merge branch 'staging' into feature/world-schedule

alexey-petrov commented on src/components/organisms/AdminVenueView/components/EventsView/EventsView.tsx:
```diff
@@ -112,20 +123,24 @@ export const EventsView: React.FC<EventsViewProps> = ({ venueId, venue }) => {
+            <Checkbox
+              checked={showSplittedEvents}
+              onChange={toggleSplittedEvents}
+              label="Split by space"
```
let's replace with `Split by ${SPACE_TAXON.lower}`

alexey-petrov commented on src/pages/WorldSchedule/WorldSchedule.tsx:
```diff
@@ -0,0 +1,253 @@
+  const { worldSlug } = useWorldParams();
+  const { isLoaded, world } = useWorldBySlug(worldSlug);
+
+  console.log(world);
```
let's remove this

alexxpetrov reviewed (approved):
A couple of minor things to fix

colinhowe committed:
Merge pull request #2774 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2778 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2780 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2783 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2784 from sparkletown/staging

Add missing import

colinhowe committed:
Merge pull request #2786 from sparkletown/staging

Add missing import

colinhowe committed:
Merge pull request #2787 from sparkletown/staging

Do not repeat the background

denisdimitrov committed:
merge staging into feature/world-schdule

denisdimitrov committed:
remove console log

denisdimitrov committed:
add space slugs to related venues hook

denisdimitrov committed:
add proper validation for event schema

denisdimitrov committed:
replace space with taxon const

colinhowe committed:
Merge pull request #2795 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2796 from sparkletown/staging

Allow listing of users again

colinhowe committed:
Merge pull request #2798 from sparkletown/staging

Fix visibility of the RunTabUsers component (#2797)

sunny-viktoryia commented on src/api/admin.ts:
```diff
@@ -42,6 +42,20 @@ export interface EventInput {
+  durationHours: number;
+  durationMinutes?: number;
+  host: string;
+  space?: {
+    id?: string;
+    slug?: string;
+  };
```
do we need `slug` here? can't we just save `spaceId`?

denisdimitrov replied:
`slug` is not replaced with `name` we need the name to show in the dropdown but we need the `id` to be passed to the function and update the venue.

sunny-viktoryia commented on src/components/organisms/AdminVenueView/components/EventsView/EventsView.tsx:
```diff
@@ -21,23 +24,33 @@ import { Checkbox } from "components/atoms/Checkbox";
     spaceIds: relatedVenueIds,
   });
 
+  const isWorldEvent = variant === EventsVariant.world;
```
Looks like now I can see all the world experiences in all the spaces. And current space and world schedules differ only in the Checkbox "Split by Space", although we can see all the experiences in the space schedule as well.

shouldn't we filter and show the Experiences that are related to the specific space if it's the Schedule tab in the space settings? 
and if it's a world schedule, then we show all the experiences.

whoaitsaimz replied:
> shouldn't we filter and show the Experiences that are related to the specific space if it's the Schedule tab in the space settings?
> and if it's a world schedule, then we show all the experiences.

That is exactly what we are trying to achieve, yes

sunny-viktoryia reviewed (commented):
some questions and comments from me. 🙃

colinhowe committed:
Merge pull request #2800 from sparkletown/staging

Deploy

colinhowe committed:
Merge pull request #2801 from sparkletown/staging

Fix for events not being set by the time code is executed

colinhowe committed:
Merge pull request #2803 from sparkletown/staging

Revert 93379de4e4b69c5c90425753a1f884ac673eb73b

colinhowe committed:
Merge pull request #2805 from sparkletown/staging

Fixes poster video (#2804)

whoaitsaimz commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -114,6 +135,24 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
+            onClick={() => setValue("space", space)}
+            className="SpacesDropdown__item"
+          >
+            {space.slug}
```
Definitely use **names** on anything admin or attendee with very few exceptions.

(The only exceptions I can think of are: the slug input field on the admin, and URL routing. Can't think of a reason we would need it showing to the user anywhere else.)

denisdimitrov commented on src/pages/SpacesDashboard/SpacesDashboard.tsx:
```diff
@@ -112,6 +113,17 @@ export const SpacesDashboard: React.FC = () => {
                 Settings
               </ButtonNG>
             </div>
+            <ButtonNG
```
I added only the button, we can easily add the Tools section later on.

denisdimitrov commented on src/types/events.ts:
```diff
@@ -0,0 +1,4 @@
+export enum EventsVariant {
```
Good point, I say we keep it as it is for now just to keep the consistency, however it would be good to rename all of these to avoid confusion in the future.

goran-peoski-work replied:
Rename now what you can. Put into constant what you can't.
Example: name your types and variables experience, but use `COLLECTION_EXPERIENCE="events"` to pass to DB queries.

Also, please do avoid enums, they're hard to deal with since it's TS trying to add C# thing by compiling into a complex JS. The same can be accomplished by:

`export type ExperiencePlace = "world" | "space"; `

goran-peoski-work replied:
Oh, you can also make it a branded string. More info at  https://github.com/sparkletown/sparkle/blob/staging/src/types/id.ts

denisdimitrov committed:
fixes and refactoring for comments

denisdimitrov committed:
show world events and space events separately

denisdimitrov committed:
merge with stagign

denisdimitrov committed:
fix space schedule events to show up correctly

denisdimitrov committed:
merge with staging

whoaitsaimz commented on src/pages/WorldSchedule/WorldSchedule.tsx:
```diff
@@ -0,0 +1,251 @@
+    ),
+  startTime: Yup.string()
+    .required()
+    .test("startTime", "Start time must be before the end time", testEndTime),
```
If it's reasonable to make this text change...

Global ending time
~~"End date must not be before the starting date"~~ negative/inverse language... use instead: 
"End date must be after the start date"

denisdimitrov commented:
@sunny-viktoryia I just fixed all of your comments, had to merge with staging twice and I had quite a lot of conflicts and another quick review of the code, wouldn't hurt.

P.S. Please ignore the big package-lock.json change, I'm looking into it atm, it was probably my fault in one of the merges. :)

denisdimitrov committed:
fix package-lock.json

denisdimitrov committed:
Merge branch 'staging' into feature/world-schedule

goran-peoski-work commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -18,84 +17,102 @@ import { useRelatedVenues } from "hooks/useRelatedVenues";
-  template?: VenueTemplate;
-  venue: WithId<AnyVenue>;
+  venueId?: string;
   setEditedEvent: Function | undefined;
```
[code smell] usually handlers are like `onEventEdit` and usually the `Function` is discouraged in favor of `(args, here)=>result` kind of type


```suggestion
  setEditedEvent: Function | undefined;
```

denisdimitrov replied:
I completely agree this is old code and should be fixed.

goran-peoski-work reviewed (changes requested):
Few suggestions

denisdimitrov commented on src/components/organisms/AdminVenueView/AdminVenueView.tsx:
```diff
@@ -44,7 +44,7 @@ import "./AdminVenueView.scss";
 
 export enum AdminVenueTab {
   spaces = "spaces",
-  timing = "timing",
+  schedule = "schedule",
   run = "run",
 }
```
I think `enum` is in favor here because we are also using it for values as well. Ex: `AdminVenueTab.run` returns `"run"`

denisdimitrov commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -18,84 +17,102 @@ import { useRelatedVenues } from "hooks/useRelatedVenues";
+  venueId?: string;
   setEditedEvent: Function | undefined;
   setShowDeleteEventModal: () => void;
+  worldId?: string;
```
The `Space` interface has `worldId: string` and `space.worldId` is passed to this prop. Changing it here to `WorldId` forces me to change it in the `Space` interface as well, which then also forces me to change the prop types in another 20 usages, which is too much imo for this PR.

goran-peoski-work replied:
This is the reason why I did some changes as `SpaceId | string` to be noticeable we'll want to move to the branded string in the future, but it requires too many changes.

denisdimitrov committed:
fixes and changes

denisdimitrov committed:
fix types and values

goran-peoski-work commented on src/components/organisms/AdminVenueView/components/EventsView/EventsView.tsx:
```diff
@@ -19,24 +23,36 @@ import { Checkbox } from "components/atoms/Checkbox";
-  spaceId: SpaceId | string;
-  space: SpaceWithId;
+export type EventsViewProps = {
+  spaceId?: SpaceId | string;
+  space?: SpaceWithId;
```
Hm... will these have different space ID values?

goran-peoski-work commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -18,84 +17,102 @@ import { useRelatedVenues } from "hooks/useRelatedVenues";
     mode: "onSubmit",
     reValidateMode: "onChange",
     validationSchema: eventEditSchema,
+    validationContext: {
+      eventSpaceId,
+    },
```
Interesting little schema :)

goran-peoski-work commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -18,84 +17,102 @@ import { useRelatedVenues } from "hooks/useRelatedVenues";
           .unix(event.startUtcSeconds)
           .format(DAYJS_INPUT_DATE_FORMAT),
-        start_time: dayjs
+        startTime: dayjs
```
[OOS] this reminds me, we still haven't replaced  `dayjs`

goran-peoski-work commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -105,23 +122,51 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
+                {renderedSpaceIds}
+              </Dropdown>
+              {errors.spaceId && (
+                <span className="input-error">{errors.spaceId.message}</span>
```
[NOTE] to self: eradicate `input-error` classes

goran-peoski-work commented on src/components/organisms/TimingEventModal/TimingEventModal.tsx:
```diff
@@ -194,75 +239,111 @@ export const TimingEventModal: React.FC<TimingEventModalProps> = ({
+              </label>
+              <div className="TimingEventModal__container">
+                {/*
+                  These wrapper divs are here to unmuddle some precedance issues
```
`@debt` tag?

goran-peoski-work commented on src/pages/WorldSchedule/WorldSchedule.tsx:
```diff
@@ -0,0 +1,193 @@
+    const {
+      inputFormattedDateSegment: startDate,
+      inputFormattedTimeSegment: startTime,
+    } = convertDateFromUtcSeconds(world?.startTimeUnix ?? NaN);
```
Interesting approach `?? NaN`.

Could be useful to have a comment explaining how `null` or `undefined` just don't work but `NaN` does.

denisdimitrov replied:
It was copy pasted from the regular schedule, good thing that you pointed it out. The reason it because `NaN` is actually a number. But maybe we should just return `0` instead.

goran-peoski-work reviewed (approved):
LGTM

Could use a code comment or two

denisdimitrov committed:
add additional fixes and twaeks

colinhowe committed:
New video API ported to master (#2854)

* New APIs for video calls (#2835)

This fully replaces all the previous hook-based code for making video
calls with simpler code that uses a single instance of a Twilio Client
and uses callbacks to manage state.

* Reinstate reactions for new video comms

* Set error rather than throwing

colinhowe committed:
Fix reaction styles for auditorium

colinhowe committed:
Do not deploy master to 7

denisdimitrov committed:
fixes the dropdown in world schedule

denisdimitrov committed:
merge with staging

denisdimitrov commented:
This is stale and very old design. It's unnecessary.
